src/main/java/com/hartwig/snpcheck/SnpCheck.java

```diff
@@ -66,51 +60,71 @@ public SnpCheck(final RunApi runs, final SampleApi samples, final Bucket snpchec
-                LOGGER.info("Passing through event for sample [{}]", event.pipeline().sample());
-                publishValidated(event);
-            } else if (run.getIni().equals(Ini.SOMATIC_INI.getValue()) || run.getIni().equals(Ini.SINGLESAMPLE_INI.getValue())) {
-                LOGGER.info("Received a SnpCheck candidate [{}] for run [{}]", run.getSet().getName(), run.getId());
-                if (run.getStatus() == Status.FINISHED || runFailedQc(run)) {
-                    Iterable<Blob> valVcfs = Optional.ofNullable(snpcheckBucket.list(Storage.BlobListOption.prefix(SNPCHECK_VCFS)))
-                            .map(Page::iterateAll)
-                            .orElse(Collections.emptyList());
-                    Optional<Sample> maybeRefSample = onlyOne(samples, run.getSet(), SampleType.REF);
-                    Optional<Sample> maybeTumorSample = onlyOne(samples, run.getSet(), SampleType.TUMOR);
-                    if (maybeRefSample.isPresent()) {
-                        Sample refSample = maybeRefSample.get();
-                        Optional<Blob> maybeValVcf = findValidationVcf(valVcfs, refSample);
-                        if (maybeValVcf.isPresent()) {
-                            VcfComparison.Result result = doComparison(run, refSample, maybeValVcf.get());
-                            SnpCheckEvent.builder()
-                                    .publisher(turquoiseTopicPublisher)
-                                    .sample(maybeTumorSample.map(Sample::getName).orElse(refSample.getName()))
-                                    .result(result.name().toLowerCase())
-                                    .build()
-                                    .publish();
-                            if (result.equals(Result.PASS)) {
-                                publishValidated(event);
-                            }
-                        } else {
-                            LOGGER.info("No validation VCF available for set [{}].", run.getSet().getName());
-                            labPendingBuffer.add(event);
-                        }
-                    } else {
-                        LOGGER.warn("Set [{}] had no ref sample available in the API. Unable to locate validation VCF.",
-                                run.getSet().getName());
-                        failed(run, RunFailure.TypeEnum.TECHNICALFAILURE);
+        Run run = runs.get(event.pipeline().runId());
+        if (passthru) {
+            LOGGER.info("Passing through event for sample [{}]", event.pipeline().sample());
+            publishValidated(event);
+        } else if (run.getIni().equals(Ini.SOMATIC_INI.getValue()) || run.getIni().equals(Ini.SINGLESAMPLE_INI.getValue())) {
+            LOGGER.info("Received a SnpCheck candidate [{}] for run [{}]", run.getSet().getName(), run.getId());
+            if (event.pipeline().context().equals(Pipeline.Context.RESEARCH)) {
+                passthroughResearchWhenDiagnosticValidated(event, run);
```

Comment:
The use the passthrough for both a configured state and a calculated state may or may not be confusing. It's not really right now but with the passage to time it may become moreso.

Also can we make the spelling consistent if we decide to keep it (I guess it's really just the flag on line 64 that is different)?